Dynatrace/dynatrace-operator
# Removes "alpha." from the feature flags + adds deprecation warning for old flags

# Description

Feature flag shouldn't have the `alpha.` prefix, because they don't mean anything.
And moving them to `beta` or something else will not improve them just introduce more complex logic in the code.

If we want to deprecate 1 or a set of feature-flags we can do it using the validation webhook. 


But because some user might already use some of these we should support them for a release or two, so this PR enables the use of both.
There is an exception when it comes to the new `statsd` feature-flags as they were not released before, so we don't have to support it there.

Also adds a warning in the webhook, if there is a deprecated annotation present in the annotation of the dynakube.

## How can this be tested?
Try to apply a dynakube, with a deprecated feature-flag, it should work but you should also get a warning.
Then do the same but without the `alpha.` part, it should still work, but without the warning on the apply


## Checklist
